app/serializers/dist/json_serializer.rb

```diff
@@ -70,5 +72,61 @@ def serialize_nested(connection)
         name: connection.name,
       }
     end
+
+    def build_mapping_blocks(mapping_rules)
+      all_mapping_blocks = []
+      puts "Generating mappings ..."
+      mapping_count = 0
+      Integration.all.each do |integration|
+        [true, false].each do |from_shopify|
+          mapping_rule_block = mapping_rules.where(
+            integration_id: integration.id,
+            from_shopify: from_shopify,
+          )
+          next if mapping_rule_block.count.zero?
+
+          mappings = []
+          Category.verticals.each do |vertical|
+            mappings << MappingBuilder.build_one_to_one_mappings_for_vertical(
+              mapping_rules: mapping_rule_block,
+              vertical: vertical,
+            )
+          end
+          processed_mappings = mappings.flatten.compact
+          mapping_count += processed_mappings.count
+          all_mapping_blocks << {
+            input_taxonomy: "shopify/v1",
+            output_taxonomy: "#{integration.name}/v1",
+            rules: processed_mappings,
+          }
+        end
+      end
+      puts "✓ Generated #{mapping_count} mappings"
+      all_mapping_blocks
+    end
+
+    def serialize_mapping_block(mapping_block)
+      {
+        input_taxonomy: mapping_block[:input_taxonomy],
+        output_taxonomy: mapping_block[:output_taxonomy],
+        rules: mapping_block[:rules]&.map(&method(:serialize_mapping)),
+      }
+    end
+
+    def serialize_mapping(mapping)
+      mapping[:input][:product_category_id] = Category.find(mapping[:input][:product_category_id]).gid
+      if mapping[:input][:attributes].present?
+        mapping[:input][:attributes] = mapping[:input][:attributes].map do |attribute|
+          {
+            name: Property.find(attribute[:name]).gid,
+            value: attribute[:value].nil? ? nil : PropertyValue.find(attribute[:value]).gid,
```

Comment:
```suggestion
            value: attribute[:value] || PropertyValue.find(attribute[:value]).gid,
```

Reply:
I have to revert the change here because in some cases a nil attribute value will lead to a specific output that has been defined in YAML mapping rules.

Reply:
Ah fair. Kinda makes me want a `NilPropertyValue` instead of magic nil, but this is fine